### PR TITLE
Receiver <-> gateway: change statistics JSON structure

### PR DESCRIPTION
**Pull request by janpriessnitz**

### CODE REVIEW

**Comment by jblomer** [comment]
LGTM But we should add a unit test for the generated statistics JSON snippet

File: cvmfs/statistics.cc [hunk 1]
```diff
@@ -91,17 +91,52 @@ string Statistics::PrintList(const PrintOptions print_options) {
...
+  JsonStringGenerator json_statistics;
+
+  // Make use of std::map key ordering and add counters namespace by namespace
+  JsonStringGenerator json_statistics_namespace;
+  std::string last_namespace = "";
   for (map<string, CounterInfo *>::const_iterator i = counters_.begin(),
                                                   iEnd = counters_.end();
        i != iEnd; ++i) {
-    json_generator.AddUnquoted(i->first, i->second->counter.ToString());
+    size_t dot_pos = i->first.find_first_of(".");
```
Comment (jblomer) [inline cvmfs/statistics.cc:117]: You can use `SplitString` from the cvmfs string utilities that gives you all the tokens in one go

**Comment by jblomer** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@cernvm-bot cloudtest

**Comment by janpriessnitz** [comment]
@jblomer This PR should be ready for merging. Failed tests are unrelated and gateway already has the corresponding change: https://github.com/cvmfs/cvmfs-gateway/commit/7454892ea4c357c7c9673b39d5f03bba33f525c5
Although, the change is only present in master branch!

**Pull request merged**